server/CGameHandler.cpp

```diff
@@ -5317,11 +5235,10 @@ bool CGameHandler::handleDamageFromObstacle(const CStack * curStack, bool stackI
 {
 	if(!curStack->alive())
 		return false;
-	bool containDamageFromMoat = false;
 	bool movementStopped = false;
 	for(auto & obstacle : getAllAffectedObstaclesByStack(curStack, passed))
 	{
-		if(obstacle->obstacleType == CObstacleInstance::SPELL_CREATED)
+		if(obstacle->obstacleType == CObstacleInstance::SPELL_CREATED || obstacle->obstacleType == CObstacleInstance::MOAT)
```

Comment:
Is this check redundant now?
Other than spell created and now moats other obstacles are non-enterable - they block battlefield & can't affect stacks

Reply:
@IvanSavenko what about Hota Dunes obstacles? It should not be a spell created AFAIK, but should be passable and should stop movement.

Reply:
@IvanSavenko there is one difference - moats are static, but Hota dunes are random, Moat for now cannot do that.

Reply:
Done (but not for Hota dunes).